### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by david-shearon**

### CODE REVIEW

**Review by justincdavis — COMMENTED** [review]
Probably change the outer most folder name to "containers", see note about the newline and question on the README running instructions.

**Comment by justincdavis** [comment]
@carlsondev @david-shearon Should the workflows be updated to use this docker container now? Is that possible to do without publishing to the Docker container hub thing?

**Comment by david-shearon** [comment]
Yes, it is possible. Maybe look here for reference: https://github.com/JonasVautherin/px4-gazebo-headless/tree/master/.github/workflows

**Comment by carlsondev** [comment]
I was referring to the integration tests workflow, which requires having a MAV SDK server instance running alongside it, not the building and deploying of the docker container. This requires starting up the docker container as a service and establishing a network. It shouldn't be that difficult, but in practice, I haven't had any luck. 

@justincdavis All the other containers don't need it because they don't need all the extra PX4 stuff and they just need a base ubuntu.

**Comment by justincdavis** [comment]
@david-shearon Can we actually update this container to use the .devcontainer framework that Github supports?

@carlsondev This could fix some of your issues as well.

https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers

**Comment by david-shearon** [comment]
@justincdavis I'll look into it. Does that just entail changing the name of the directory?

**Comment by justincdavis** [comment]
@david-shearon That and adding a docker.json or similar file (see the link). No changes to the Dockerfile are required

**Comment by justincdavis** [comment]
@david-shearon @carlsondev I think this blogpost thing could be quite useful :)

https://bradyjoslin.com/blog/dev-action-container

**Comment by david-shearon** [comment]
@justincdavis, @carlsondev  helped me with the dev container. Is this what you were looking for? It doesn't build the docker container for some reason.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Author: Jonas Vautherin
+# Original Found at: https://github.com/JonasVautherin/px4-gazebo-headless
+# Edited by: David Shearon
+
+FROM ubuntu:18.04
```
Comment (justincdavis) [inline .devcontainer/Dockerfile:5]: We should really change this to ubuntu:20.04 I guess I missed this before

```suggestion
FROM ubuntu:20.04
```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "build": { "dockerfile": "Dockerfile" },
+  
+    "customizations": {
+      "vscode": {
+        "extensions": ["dbaeumer.vscode-eslint"]
+      }
+    },
+  
+    "forwardPorts": [14540, 14550]
+}
```
Comment (justincdavis) [inline .devcontainer/devcontainer.json:11]:
```suggestion
}

```

**Review by justincdavis — CHANGES_REQUESTED** [review]
As far as I can tell this looks like exactly what we want. I am confused why it would not be building since you essentially should have just renamed the folder to .devcontainer and added that .json file?

If you can send me the errors LMK and also not the ubuntu 20.04 change, that is what I have been using I never realized the Docker used that,

**Review by justincdavis — APPROVED** [review]

**Pull request merged**